deepchecks/base/dataset.py

```diff
@@ -122,28 +117,41 @@ def __init__(
         self._data = df.copy()
 
         # Validations
-        if label is not None:
-            if label.shape[0] != self._data.shape[0]:
-                raise DeepchecksValueError('Number of samples of label and data must be equal')
-            if len(label.shape) > 1 and label.shape[1] != 1:
-                raise DeepchecksValueError('Label must be a column vector')
-            # Make tests to prevent overriding user column
-            if label_name is None:
-                label_name = 'target'
-                if label_name in self._data.columns:
-                    raise DeepchecksValueError(f'Data has column with name "{label_name}", use label_name parameter'
-                                               ' to set column name for label which does\'t exists in the data')
-            else:
-                if label_name in self._data.columns:
-                    raise DeepchecksValueError('Can\'t pass label with label_name that exists in the data. change '
-                                               'the label_name parameter')
-
-            # If passed label is a pandas object, check that indexes match, else set column as is with provided values
-            if isinstance(label, (pd.Series, pd.DataFrame)):
-                pd.testing.assert_index_equal(self._data.index, label.index)
-                self._data[label_name] = label
+        if label is None:
+            label_name = None
+        else:
+            if isinstance(label, (pd.Series, pd.DataFrame, np.ndarray)):
+                # Validate label shape
+                if label.shape[0] != self._data.shape[0]:
+                    raise DeepchecksValueError('Number of samples of label and data must be equal')
+                if len(label.shape) > 1 and label.shape[1] != 1:
+                    raise DeepchecksValueError('Label must be a column vector')
+
+                # Get label name
+                if isinstance(label, (pd.Series, pd.DataFrame)) and label.name is not None:
+                    label_name = label.name
+                    if label_name in self._data.columns:
+                        raise DeepchecksValueError(f'Data has column with name "{label_name}", use pandas rename to'
+                                                   f' change label name or remove the column from the dataframe')
+                else:
+                    label_name = 'target'
```

Comment:
should be written in parameter documentation?
that we take the name of the series / column in dataframe, and if it's a numpy array / no name then we call it 'target'?

Reply:
added